util/reggen/validate.py

```diff
@@ -1365,16 +1370,29 @@ def make_intr_regs(regs, offset, addrsep, fullwidth):
         return iregs, 1
 
     iregs.append(
-        make_intr_reg(regs, 'INTR_STATE', offset, 'rw1c', 'hrw',
-                      'Interrupt State Register'))
+        make_intr_alert_reg(regs, 'INTR_STATE', offset, 'rw1c', 'hrw',
+                            'Interrupt State Register'))
     iregs.append(
-        make_intr_reg(regs, 'INTR_ENABLE', offset + addrsep, 'rw', 'hro',
-                      'Interrupt Enable Register'))
+        make_intr_alert_reg(regs, 'INTR_ENABLE', offset + addrsep, 'rw', 'hro',
+                            'Interrupt Enable Register'))
     iregs.append(
-        make_intr_reg(regs, 'INTR_TEST', offset + 2 * addrsep, 'wo', 'hro',
-                      'Interrupt Test Register'))
+        make_intr_alert_reg(regs, 'INTR_TEST', offset + 2 * addrsep, 'wo', 'hro',
+                            'Interrupt Test Register'))
     return iregs, 0
 
+def make_alert_regs(regs, offset, addrsep, fullwidth):
+    alert_regs = []
+    alerts = regs['alert_list']
+    if sum([int(x['width'], 0) if 'width' in x else 1
+            for x in alerts]) > fullwidth:
```

Comment:
Nit: that's hard to read as the outer brackets/sum is wrapped. Maybe it could just be turned around:
```
if fullwidth <
    sum(...)
```
or the wrapping is moved:
``` 
if sum(...)
    > fullwidth
```

Reply:
Alternatively, you could rewrite as:
```python
    if sum([int(x.get('width', '1'), 0) for x in alerts]) > fullwidth:
```
which is probably a bit simpler.

Reply:
Yes let's simplify that as you suggested.